packages/notifi-frontend-client/lib/NotifiFrontendClient.ts

```diff
@@ -73,4 +91,49 @@ export class NotifiFrontendClient {
 
     await Promise.all([saveAuthorizationPromise, saveRolesPromise]);
   }
+
+  async beginLoginViaTransaction({
+    walletBlockchain,
+    walletPublicKey,
+  }: BeginLoginProps): Promise<BeginLogInByTransactionResult> {
+    const { dappAddress } = this._configuration;
+
+    const result = await this._service.beginLogInByTransaction({
+      walletAddress: walletPublicKey,
+      walletBlockchain: walletBlockchain,
+      dappAddress,
+    });
+
+    const nonce = result.beginLogInByTransaction.nonce;
+
+    clientRandomUuid = await window.crypto.randomUUID();
+
+    return { nonce };
+  }
+
+  async completeLoginViaTransaction({
+    walletBlockchain,
+    walletPublicKey,
+    input,
+  }: BeginLoginProps &
+    CompleteLoginProps): Promise<CompleteLogInByTransactionMutation> {
+    const { transactionSignature } = input;
+    const { dappAddress } = this._configuration;
+
+    if (!clientRandomUuid) {
```

Comment:
i prefer to totally avoid truthiness checks when possible. If we change this to `string | null` we can have an explicit `!== null`